pkg/kvstore/consul.go

```diff
@@ -361,6 +354,59 @@ func (c *ConsulClient) GASNewL3n4AddrID(basePath string, baseID uint32, lAddrID
 	}
 }
 
+// StartWatch starts watching for changes in a prefix
+func (c *ConsulClient) StartWatch(w *Watcher) {
+	go func(w *Watcher) {
```

Comment:
No need to pass a Watcher parameter to this closure, it can already access the w parameter to StartWatch.